### PR TITLE
[GPU][DG2] Fix unit tests workaround

### PR DESCRIPTION
**Pull request by tuxedcat**
### Details:
 - Moving `_config.apply_user_properties(_engine.get_device_info());` from network.cpp(#15217) to program.cpp(#15800) cause many unit tests fails. Ideally it's recommended to fix such fails, but revert such modification for now.

### Tickets:
 - 67491

### CODE REVIEW

File: src/plugins/intel_gpu/src/graph/program.cpp [hunk 1]
```diff
@@ -1264,6 +1261,10 @@ void program::dump_program(const char* stage,
                            bool with_full_info,
                            std::function<bool(program_node const&)> const& filter) const {
     std::string path = get_dir_path(_config);
+    GPU_DEBUG_GET_INSTANCE(debug_inst);
+    GPU_DEBUG_IF(debug_inst->dump_graphs.size()) {
+        path = debug_inst->dump_graphs;
+    }
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/graph/program.cpp:1267]: is this still necessary?

Comment (tuxedcat) [reply src/plugins/intel_gpu/src/graph/program.cpp:1267]: No, it doesn't necessary. Removed.

File: src/plugins/intel_gpu/tests/test_utils/test_utils.cpp [hunk 2]
```diff
@@ -306,8 +306,12 @@ cldnn::engine& get_test_engine() {
 cldnn::stream_ptr get_test_stream_ptr() {
     static std::shared_ptr<cldnn::stream> test_stream = nullptr;
     if (!test_stream) {
+#ifdef ENABLE_ONEDNN_FOR_GPU
+        ExecutionConfig cfg(ov::intel_gpu::queue_type(QueueTypes::in_order));
```
Comment (isanghao) [inline src/plugins/intel_gpu/tests/test_utils/test_utils.cpp:310]: @byungilm  is this OK? I think this is not a correct fix..

Comment (byungilm) [reply src/plugins/intel_gpu/tests/test_utils/test_utils.cpp:310]: Removed. I think setting for in_order should move into each test-case.

**Comment by byungilm** [comment]
This issue moves to https://github.com/openvinotoolkit/openvino/pull/16295

**Pull request closed without merging**